crates/wascc-provider/src/lib.rs

```diff
@@ -437,21 +498,40 @@ struct VolumeBinding {
 fn wascc_run_http(
     host: Arc<Mutex<WasccHost>>,
     data: Vec<u8>,
-    env: EnvVars,
+    _env: EnvVars,
     volumes: Vec<VolumeBinding>,
     log_path: &Path,
     status_recv: Receiver<ContainerStatus>,
+    port_assigned: i32,
 ) -> anyhow::Result<ContainerHandle<ActorHandle, LogHandleFactory>> {
     let mut caps: Vec<Capability> = Vec::new();
 
+    let mut env_map: HashMap<String, String> = HashMap::new();
+    env_map.insert("PORT".to_string(), port_assigned.to_string());
     caps.push(Capability {
         name: HTTP_CAPABILITY,
         binding: None,
-        env,
+        env: env_map, //replaced _env
     });
     wascc_run(host, data, &mut caps, volumes, log_path, status_recv)
 }
 
+fn find_available_port(port_set: &Arc<Mutex<HashSet<i32>>>) -> i32 {
```

Comment:
I notice here that you are returning an `i32` with a negative integer in the case of an error. Can you please have a look at [Result types](https://doc.rust-lang.org/std/result/) in Rust to return errors?

Reply:
What do you think about writing some unite tests to test some of the common cases this function covers?

cc @thomastaylor312, do you think we need to write some unit tests before merging this code?